src/hooks/useWrapCallback.ts

```diff
@@ -17,58 +19,117 @@ export enum WrapType {
   UNWRAP,
 }
 
+export enum WrapState {
+  UNKNOWN,
+  PENDING,
+  WRAPPED,
+  UNWRAPPED,
+}
+
 const NOT_APPLICABLE = { wrapType: WrapType.NOT_APPLICABLE }
+interface UseWrapCallbackReturn {
+  wrapType: WrapType
+  execute?: () => Promise<void>
+  inputError?: string
+  wrapState?: WrapState
+  setWrapState?: (wrapState: WrapState) => void
+}
+
 /**
  * Given the selected input and output currency, return a wrap callback
  * @param inputCurrency the selected input currency
  * @param outputCurrency the selected output currency
  * @param typedValue the user input value
  */
-export default function useWrapCallback(
-  inputCurrency: Currency | undefined,
-  outputCurrency: Currency | undefined,
-  typedValue: string | undefined
-): { wrapType: WrapType; execute?: undefined | (() => Promise<void>); inputError?: string } {
+export function useWrapCallback(
+  currencies: { INPUT?: Currency | undefined; OUTPUT?: Currency | undefined },
```

Comment:
This seems unnecessary IMHO. This pattern was introduced by Uniswap, but it's weird. 

```
  inputCurrency: Currency | undefined,
  outputCurrency: Currency | undefined,
````

Is just cleaner. All caps signals constants, but this is just a mere object. :)

Reply:
ok, I think I understand. Reverted.